torch/optim/rprop.py

```diff
@@ -300,11 +300,10 @@ def _view_complex_as_real(tensor_list):
         # At the end of the step, grouped_prevs will contain the current grads, so we reuse
         # grouped_prevs memory instead of creating a new buffer, but, for clarity, we reassign
         # to keep referring to the buffer as grouped_grads.
-        for i in range(len(grouped_prevs)):
-            if maximize:
-                grouped_prevs[i].copy_(grouped_grads[i] * -1)
-            else:
-                grouped_prevs[i].copy_(grouped_grads[i])
+        if maximize:
+            torch._foreach_copy_(grouped_prevs, torch._foreach_neg(grouped_grads))
+        else:
+            torch._foreach_copy_(grouped_prevs, grouped_grads)
```

Comment:
```suggestion
        torch._foreach_copy_(grouped_prevs, grouped_grads)
        if maximize:
            torch._foreach_neg_(grouped_prevs)            
```
to save on intermediate memory